File: 15_1counts_in_n/main.py
# -*- coding: utf-8 -*-


def number_of_1(n):
    if n < 0:
        n &= 0xffffffff
    count = 0
    while n:
        count += 1
        n &= n - 1
    return count


if __name__ == '__main__':
    print(number_of_1(9))

File: 15_1counts_in_n/main.cpp
#include <iostream>

int NumberOf1(int n) {
    int count = 0;
    while (n) {
        n &= n - 1;
        ++count;
    }
    return count;
}

int main() {
    std::cout << NumberOf1(-2) << std::endl;
}
